File: 02. C# Part2/06. StringsАndTextProcessing-Homework/12. ParseURL/ParseURL.cs
using System;
using System.Linq;

//Write a program that parses an URL address given in the format: [protocol]:
//[server]/[resource] and extracts from it the [protocol], [server] and [resource] elements.

    class ParseURL
    {
        static void Main()
        {
            Console.WriteLine("Please enter an URL: ");
            string input = Console.ReadLine();
            int protocolEnd = input.IndexOf(':');
            string protocol = input.Substring(0, protocolEnd);
            int serverEnd = input.IndexOf('/', 8);
            int serverStart = input.IndexOf('/');
            string server = input.Substring(serverStart + 2, serverEnd - serverStart - 2);
            string resource = input.Substring(serverEnd);
            Console.WriteLine("[protocol] = {0}", protocol);
            Console.WriteLine("[server] = {0}", server);
            Console.WriteLine("[resource] = {0}", resource);
        }
    }

File: 06. JavaScript Basics/08. Strings-Homework/08. Strings-Homework/07. ParseURL/ParseURL.js
//Write a script that parses an URL address given in the format: [protocol]://[server]/[resource] 
//and extracts from it the [protocol], [server] and [resource] elements.
//Return the elements in a JSON object.

function parseURL(urlAddress) {
    function extractURLElements(url) {
        var pattern = /(.*):\/\/(.*?)(\/.*)/g;

        return pattern.exec(url);
    }


    var fragments = extractURLElements(urlAddress);

    // Convert to JSON object
    var jsonObject = JSON.stringify({
        protocol: fragments[1],
        server: fragments[2],
        resource: fragments[3]
    });


    // Parse JSON object to JS object
    var jsObject = JSON.parse(jsonObject);

    jsConsole.writeLine("URL Address: " + urlAddress);
    jsConsole.writeLine("[protocol] = " + jsObject.protocol);
    jsConsole.writeLine("[server] = " + jsObject.server);
    jsConsole.writeLine("[resource] = " + jsObject.resource);
}
parseURL('http://telerikacademy.com/Courses/Courses/Details/239');